src/navigation/navigation-events.js

```diff
@@ -58,7 +58,7 @@ export function navigateToUrl(obj) {
     window.location.hash = destination.hash;
   } else {
     // different path, host, or query params
-    window.history.pushState(null, null, url);
+    window.history.pushState({}, '', url);
```

Comment:
I think this code might overwrite `history.state` to `{}` rather than keeping it at the previous value. I don't know if `null` is any better, but would like to fully understand the implications of this change and ideally see tests for it.